### ISSUE TITLE
Operator cancels rollout of ds/deployment

### ISSUE THREAD

**Issue by andrewdinunzio** [open-telemetry/opentelemetry-operator#1300]
I can't say for sure this is what's happening, but it seems like it to me. `kubectl rollout restart` restarts the workload by adding/modifying an annotation to the pod template spec in order to trigger a rollout. It seems like when it does this, the controller is immediately reconciling and removing the new annotation. I assume this is because it's replacing the annotations field instead of doing some kind of merge patch / SSA.

This causes the OpenTelemetry Collector DaemonSet to restart maybe one pod, but the rollout stops after that.

**Comment by pavolloffay** [comment]
If that is the case, shouldn't the `kubectl rollout restart` be called on the OTELcol CRD?

**Comment by andrewdinunzio** [comment]
I don't think that's possible. Just tried and got this error:
```
$ kubectl rollout restart otelcol otel
error: no kind "OpenTelemetryCollector" is registered for version "opentelemetry.io/v1alpha1" in scheme "pkg/scheme/scheme.go:28"
```

I think the things that are "rollout"-able are limited to Deployments, DaemonSets, and StatefulSets. Not sure if there's a way to configure that.

**Comment by pavolloffay** [comment]
Anything that changes the collector pods spec in deployment will cause a restarts - e.g. probably adding a new annotation https://github.com/open-telemetry/opentelemetry-operator/blob/d3c98ae529fee23ccc92c6722ca2141406a3dc40/apis/v1alpha1/opentelemetrycollector_types.go#L134

**Comment by andrewdinunzio** [comment]
Yes, doing a rollout on the deployment should do that, but the controller immediately replaces the annotations back to the way it was, which seemingly cancels the rollout

**Comment by andrewdinunzio** [comment]
Oh I see. That's a good workaround.

**Comment by verejoel** [comment]
I think this needs to be tackled properly, not worked around. This is problematic, for example, when spot instances are taken down. We have seen pods on non-existent nodes, and rollouts stuck because of this.

How would a fix look? Naively I'd say we just exclude the rollout annotation from reconciliation. Would be happy to work on it.

**Comment by jaronoff97** [comment]
@verejoel that would be great. Thank you!

**Comment by jaronoff97** [comment]
@verejoel im going to reassign this one, as its something i just ran into. Let me know if you've already begun. If not @Toaddyan take a look

**Comment by verejoel** [comment]
Please go ahead, I don’t have time to look at this too deeply at the moment

**Comment by Toaddyan** [comment]
So I was working on trying to re-produce the issue: 
```
➜  otel-collector-charts git:(main) ✗ k get pods
NAME                                                       READY   STATUS    RESTARTS      AGE
kube-otel-stack-kube-state-metrics-66b58457bf-7bq9l        1/1     Running   1 (11m ago)   24m
kube-otel-stack-metrics-collector-0                        1/1     Running   1 (11m ago)   22m
kube-otel-stack-metrics-collector-4vfrt                    1/1     Running   0             2m22s
kube-otel-stack-metrics-collector-zh75g                    1/1     Running   0             2m22s
kube-otel-stack-metrics-targetallocator-6fb775b699-bhdbp   1/1     Running   1 (11m ago)   24m
kube-otel-stack-prometheus-node-exporter-4v8rh             1/1     Running   0             8m54s
kube-otel-stack-prometheus-node-exporter-j7lpm             1/1     Running   1 (11m ago)   24m
opentelemetry-operator-88dc494dc-6f5kp                     2/2     Running   2 (11m ago)   29m
➜  otel-collector-charts git:(main) ✗ k rollout restart daemonset kube-otel-stack-metrics-collector
daemonset.apps/kube-otel-stack-metrics-collector restarted
➜  otel-collector-charts git:(main) ✗ k get pods 
NAME                                                       READY   STATUS    RESTARTS      AGE
kube-otel-stack-kube-state-metrics-66b58457bf-7bq9l        1/1     Running   1 (11m ago)   24m
kube-otel-stack-metrics-collector-0                        1/1     Running   1 (11m ago)   22m
kube-otel-stack-metrics-collector-hvs7b                    1/1     Running   0             3s
kube-otel-stack-metrics-collector-mdlm2                    1/1     Running   0             3s
kube-otel-stack-metrics-targetallocator-6fb775b699-bhdbp   1/1     Running   1 (11m ago)   24m
kube-otel-stack-prometheus-node-exporter-4v8rh             1/1     Running   0             9m8s
kube-otel-stack-prometheus-node-exporter-j7lpm             1/1     Running   1 (11m ago)   24m
opentelemetry-operator-88dc494dc-6f5kp                     2/2     Running   2 (11m ago)   30m
```
From the above conversation, I'm interpreting that upon a rollout restart of a type of resource (daemonset), I should see the problem of a subset of pods restarting, where I SHOULD be seeing ALL of the pods restarting.
---
In my recent run down of this, I'm not seeing the OP's stated behavior of restarts. 
---
am I missing anything particular here?

**Comment by jaronoff97** [comment]
Similar to @Toaddyan, I tried to repro this and was unable to. I think this may have been resolved in #1995 inadvertently. I'm going to close this for now. Please let me know if I should re-open this!
